terra_layer/models.py

```diff
@@ -216,6 +216,33 @@ def save(self, **kwargs):
     def __str__(self):
         return f"Layer({self.id}) - {self.name}"
 
+    @transaction.atomic()
+    def replace_source(self, source, fields_matches={}):
+        # update old field if ones from the new source
+        # remove it when not present in the new source
+        for filter_field in self.fields_filters.all():
+
+            if source.fields.filter(name=filter_field.field.name).exists():
+                filter_field.field = source.fields.get(name=filter_field.field.name)
+            elif source.fields.filter(
+                name=fields_matches.get(filter_field.field.name, "")
+            ).exists():
+                filter_field.field = source.fields.get(
+                    name=fields_matches.get(filter_field.field.name)
+                )
```

Comment:
```suggestion
            field_name = fields_matches.get(filter_field.field.name, filter_field.field.name)
            if source.fields.filter(name=field_name).exists():
                filter_field.field = source.fields.get(name=field_name)
```

Reply:
Since I amend the commit, I did a refactoring too.